nemo_skills/inference/server/model.py

```diff
@@ -673,37 +686,45 @@ def _generate_single(
         
         retry_count = 0
         retry_delay = self.initial_retry_delay
-        
         while True:
             try:
-                response = self.client.chat.completions.create(
-                    model=self.model,
-                    temperature=temperature,
-                    top_p=top_p,
-                    max_tokens=tokens_to_generate,
-                    presence_penalty=repetition_penalty,
-                    seed=random_seed,
-                    stop=stop_phrases,
-                    messages=prompt,
-                    logprobs=top_logprobs is not None,
-                    top_logprobs=top_logprobs,
-                    timeout=timeout,
-                )
-                break  # Success, exit the retry loop
+                if self.model_type == "Azure OpenAI":
+                    response = self.client.chat.completions.create(
+                        model=self.model,
+                        temperature=1,
```

Comment:
do they really not support temperature? I'd double check all parameters here, hopefully there is more that can be reused. And then you can just have all parameters as a dict variable and just pop/rename a few for the Azure case and have a single call to get a response